lib/metadata_fields.rb

```diff
@@ -0,0 +1,215 @@
+module MetadataFields
+	extend ActiveSupport::Concern
+	extend self
+
+	NAMESPACE_ALL = 'all'
+	MULTIPLE_ALL  = 'all'
+	DEFAULT_NAMESPACE = NAMESPACE_ALL
+	DEFAULT_MULTIPLE  = MULTIPLE_ALL
+	METADATA_FIELDS = YAML.load_file(File.join(Rails.root, "config", "metadata_fields.yml"))["terms"]["vocabulary"].deep_symbolize_keys
+	DISPLAY_CONFIG = YAML.load_file(File.join(Rails.root, "config", "metadata_fields.yml"))["config"]
+
+	# get metadata fields
+	def process_metadata_field_names(ns:DEFAULT_NAMESPACE, multiple: DEFAULT_MULTIPLE)
+		unless allowed_values_for_ns.include?(ns)
```

Comment:
@jgpawletko There's a whole bunch of these throughout the file. Do you reuse "emsg" when using that pattern, or do you have multiple variabls (emsg_ns, emsg_multi, etc)?